test/command_manager_test.cpp

```diff
@@ -0,0 +1,610 @@
+
+
+using namespace rosflight_firmware;
+
+// Initialize the full firmware, so that the state_manager can do its thing
+void step_firmware(ROSflight& rf, testBoard& board, uint32_t us)
+{
+  uint64_t start_time_us = board.clock_micros();
+  float dummy_acc[3] = {0, 0, -9.80665};
+  float dummy_gyro[3] = {0, 0, 0};
+  while(board.clock_micros() < start_time_us + us)
+  {
+    board.set_imu(dummy_acc, dummy_gyro, board.clock_micros() + 1000);
+    rf.run();
+  }
+}
+
+TEST(command_manager_test, rc) {
+  testBoard board;
+  ROSflight rf(board);
+
+  // Initialize the firmware
+  rf.init();
+
+  uint16_t rc_values[8];
+  for (int i = 0; i < 8; i++)
+  {
+    rc_values[i] = 1500;
+  }
+  rc_values[2] = 1000;
+
+  float max_roll = rf.params_.get_param_float(PARAM_RC_MAX_ROLL);
+  float max_pitch = rf.params_.get_param_float(PARAM_RC_MAX_PITCH);
+  float max_yawrate = rf.params_.get_param_float(PARAM_RC_MAX_YAWRATE);
+
+
+  //=================================================
+  // RC Commands Test
+  //=================================================
+  // First, lets just try sending rc_commands alone
+  board.set_rc(rc_values);
+  step_firmware(rf, board, 20000);
+
+  control_t output = rf.command_manager_.combined_control();
+  EXPECT_EQ(output.x.type, ANGLE);
+  EXPECT_PRETTYCLOSE(output.x.value, 0.0);
+  EXPECT_EQ(output.y.type, ANGLE);
+  EXPECT_PRETTYCLOSE(output.y.value, 0.0);
+  EXPECT_EQ(output.z.type, RATE);
+  EXPECT_PRETTYCLOSE(output.z.value, 0.0);
+  EXPECT_EQ(output.F.type, THROTTLE);
+  EXPECT_PRETTYCLOSE(output.F.value, 0.0);
+
+  rc_values[0] = 2000;
+  rc_values[1] = 1000;
+  rc_values[2] = 1500;
+  rc_values[3] = 1250;
+  board.set_rc(rc_values);
+  step_firmware(rf, board, 20000);
+
+  output = rf.command_manager_.combined_control();
+  EXPECT_EQ(output.x.type, ANGLE);
+  EXPECT_PRETTYCLOSE(output.x.value, 1.0*max_roll);
+  EXPECT_EQ(output.y.type, ANGLE);
+  EXPECT_PRETTYCLOSE(output.y.value, -1.0*max_pitch);
+  EXPECT_EQ(output.z.type, RATE);
+  EXPECT_PRETTYCLOSE(output.z.value, -0.5*max_yawrate);
+  EXPECT_EQ(output.F.type, THROTTLE);
+  EXPECT_PRETTYCLOSE(output.F.value, 0.5);
+}
+
+
+TEST(command_manager_test, rc_arm_disarm) {
+  testBoard board;
+  ROSflight rf(board);
+
+  // Make sure that rc is hooked up
+  board.set_pwm_lost(false);
+
+  // Initialize the firmware
+  rf.init();
+
+  uint16_t rc_values[8];
+  for (int i = 0; i < 8; i++)
+  {
+    rc_values[i] = 1500;
+  }
+  rc_values[2] = 1000;
+
+  float max_roll = rf.params_.get_param_float(PARAM_RC_MAX_ROLL);
+  float max_pitch = rf.params_.get_param_float(PARAM_RC_MAX_PITCH);
+  float max_yawrate = rf.params_.get_param_float(PARAM_RC_MAX_YAWRATE);
+
+  // Let's clear all errors in the state_manager
+  rf.state_manager_.clear_error(rf.state_manager_.state().error_codes);
+
+  //=================================================
+  // RC Arming Test
+  //=================================================
```

Comment:
We could do that.  I figured that they were extensive enough that they justified their own file.  They used to be a bunch of separate executables, but I combined them to make it easier to run the test.